### DISCUSSION TITLE
validation on custom lidar only dataset (indoor_eval)

### DISCUSSION THREAD

**Discussion by gynaa** [open-mmlab/mmdetection3d#2422] [Q&A]
Hello,

I have a question regarding the validation step. I am trying to train PartA2 on a custom lidar only dataset. The training works fine. However, during the validation my AP is always 0.0000. I was wondering does the evaluation in indoor_eval require images to work properly or should it also work with a lidar only dataset?

**Comment by sunjiahao1999** [comment]
`Indoor_eval` can work with lidar-only dataset. Perhaps it may be because your `pred_label` id mismatch your `gt_label` id?

**Reply by gynaa** [reply]
thank you very much for your quick response. so i just realized that dt_annos is filled with empty entires. meaning that pred is completely empty. I checked the forward training functions and in there bounding box predicitions are being made. Any idea to why no predictions are being made in the validation?

Here is my validation pipeline, if that helps: 
`eval_pipeline = [
    dict(type='LoadPointsFromFile', coord_type='LIDAR', load_dim=4, use_dim=3),
    dict(type='MultiScaleFlipAug3D', img_scale=(1, 1), pts_scale_ratio=1, flip=False,
        transforms=[
            dict(type='GlobalRotScaleTrans', rot_range=[0, 0], scale_ratio_range=[1., 1.], translation_std=[0, 0, 0]),
            dict(type='PointsRangeFilter', point_cloud_range=point_cloud_range),
            dict(type='DefaultFormatBundle3D', class_names=classes, with_label=False),
            dict(type='Collect3D', keys=['points'])])]`

**Reply by gynaa** [reply]
any help would be appreciated :)
